File: Sources/Applicazione/main/java/it/kaizenteam/app/model/TableImpl.java
/*
* Name: {TableImpl.java}
* Package: {it.kaizenteam.app.model}
* Location: {src.main.java.it.kaizenteam.model}
* Date: {data di creazione del file}
* Version: {0.01}
*
* History:
*
* =================================================================
* Version Date Programmer Changes
* =================================================================
* 0.01 GG-MM-AAAA Aessandro Moretto {creation}
* =================================================================
*
*/
package it.kaizenteam.app.model;


class TableImpl extends ChartImpl{

    static {
        //registro il tipo di grafico (DI)
        registerFactory("table", new TableFactory());
    }

    private TableImpl(String id) {
        super(id);
        //registro gli aggiornamenti (DI)
        registerUpdater("inplace", new TableInPlaceUpdater());
        registerUpdater("stream", new TableStreamUpdater());
    }

    /**
     * sets the settings to the TableImpl
     * @param object settings for a TableImpl
     */
    @Override
    public void setSettings(SettingsObject object) /*throws Exception*/{
        //TODO controllo che il tipo si quello corretto
        settings=object;
    }

    /**
     * sets the data to the TableImpl
     * @param object data for a TableImpl
     */
    @Override
    public void setData(DataObject object) /*throws Exception*/{
        //TODO controllo che il tipo si quello corretto
        data=object;
    }

    /**
     * factory class for a Table
     */
    protected static class TableFactory implements ChartFactory{
        /**
         * create a BarChartImpl
         * @param id identification for the TableImpl
         * @return an instance of TableImpl
         */
        @Override
        public ChartImpl create(String id) {
            return new TableImpl(id);
        }
    }

    /**
     * this class permits to update the data of the instance of a TableImpl
     */
    protected class TableInPlaceUpdater implements Updater{
        /**
         * Update the data of the TableImpl
         * @param object the update object for a Table
         */
        @Override
        public void update(UpdateObject object) {
            //TODO
        }
    }

    /**
     * this class permits to update the data of the instance of a TableImpl
     */
    protected class TableStreamUpdater implements Updater{
        /**
         * Update the data of the TableImpl
         * @param object the update object for a Table
         */
        @Override
        public void update(UpdateObject object) {
            //TODO
        }
    }
}

File: Sources/Norris/main/DataModel/TableImpl.js
/*
 * Name: {TableImpl.js}
 * Module: {}
 * Location: {Norris/Main/DataModel}
 * Date: {2015-05-04}
 * Version: {v0.1}
 *
 * History:
 *
 * ================================================================================
 * Version Date Programmer Changes
 * ================================================================================
 * 0.1 2015-05-04 Chiara Bigarella {creation}
 * ================================================================================
 */

var ChartImpl = require('./ChartImpl.js');

// module.exports = TableImpl;

ChartImpl.registerFactory('table' , new TableFactory());

var defaults = {
	title : '' ,
	description : 'This is a table.',
	showLegend : true,
	legendPosition : 'top-right',
	maxPoints : 5
};

function TableImpl (uid) {
	if (!(this instanceof TableImpl)) return new TableImpl(uid);
	ChartImpl.call(this, uid);
	this.setSettings(defaults);
	this.setType('table');
}

TableImpl.prototype.__proto__ = ChartImpl.prototype;

/* TableFactory ------------------------------------------------------- */

function TableFactory() {
	if(!(this instanceof TableFactory)) return new TableFactory();
}

TableFactory.prototype.create = function (uid) {
	return new TableImpl(uid);
};